### PR TITLE
ci: use `.nvmrc` for setup-node versions

### PR DESCRIPTION
**Pull request by yin1999**
### Description

ci: use `.nvmrc` for setup-node versions

### Related issues and pull requests

Reflect mdn/content#24355.

### CODE REVIEW

File: .github/workflows/pr-check_markdownlint.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
       - name: Setup Node.js environment
         uses: actions/setup-node@v3
         with:
-          node-version: 18
+          node-version-file: ".nvmrc"
```
Comment (yin1999) [inline .github/workflows/pr-check_markdownlint.yml:34]: Just did not add `- .nvmrc` to trigger this workflow when node version changed (as we have `markdown-lint.yml` which would lint all the markdown file while this workflow would not works well when there was not changed markdown file)

**Comment by fwqaaq** [comment]
Why not switch `yarn` to `pnpm`, `pnpm` has faster speed.

**Comment by yin1999** [comment]
> Why not switch `yarn` to `pnpm`, `pnpm` has faster speed.

We just have [a vote on package manager](https://github.com/orgs/mdn/discussions/108), and we may switch to NPM. But it seems that some problems were encountered during the migration process (follow mdn/yari#2326).

**Comment by fwqaaq** [comment]
> > Why not switch `yarn` to `pnpm`, `pnpm` has faster speed.
> 
> We just have [a vote on package manager](https://github.com/orgs/mdn/discussions/108), and we may switch to NPM. But it seems that some problems were encountered during the migration process (follow [mdn/yari#2326](https://github.com/mdn/yari/issues/2326)).

It seems like it hasn't started yet

**Comment by yin1999** [comment]
> It seems like it hasn't started yet

mdn/yari#6322

**Review by caugner — APPROVED** [review]

**Comment by caugner** [comment]
Thank you @yin1999! 🎉 

@fwqaaq We will probably have migrated mdn/yari to npm by end of May 2023.

**Pull request merged**